tests/integ/modin/frame/test_insert.py

```diff
@@ -768,3 +770,32 @@ def insert_op(df):
         expected_res = native_df1.join(native_df2["bar"], how="left", sort=False)
         expected_res = expected_res[["bar", "foo"]]
         assert_frame_equal(snow_res, expected_res, check_dtype=False)
+
+
+@sql_count_checker(query_count=4, join_count=6)
+def test_insert_timedelta():
+    native_df = native_pd.DataFrame({"col1": [1, 2], "col2": [3, 4]})
+    snow_df = pd.DataFrame(native_df)
+
+    def insert(column, vals, df):
+        if isinstance(df, pd.DataFrame) and isinstance(vals, native_pd.Series):
+            values = pd.Series(vals)
+        else:
+            values = vals
+        df.insert(1, column, values)
+        return df
+
+    vals = native_pd.timedelta_range(1, periods=2)
+    eval_snowpark_pandas_result(
+        snow_df, native_df, functools.partial(insert, "td", vals)
+    )
+
+    vals = native_pd.Series(native_pd.timedelta_range(1, periods=2))
+    eval_snowpark_pandas_result(
+        snow_df, native_df, functools.partial(insert, "td2", vals)
+    )
+
+    vals = native_pd.Series(native_pd.timedelta_range(1, periods=2), index=[0, 2])
+    eval_snowpark_pandas_result(
+        snow_df, native_df, functools.partial(insert, "td3", vals)
+    )
```

Comment:
```suggestion
@pytest.mark.parametrize("values,query_count",
                         [
                             (native_pd.timedelta_range(1, periods=2), 2),  (native_pd.Series(native_pd.timedelta_range(1, periods=2)), 1), (native_pd.Series(native_pd.timedelta_range(1, periods=2), index=[0, 2]), 1)])
def test_insert_timedelta(values, query_count):
    native_df = native_pd.DataFrame({"col1": [1, 2], "col2": [3, 4]})
    snow_df = pd.DataFrame(native_df)
    def insert(column, vals, df):
        if isinstance(df, pd.DataFrame) and isinstance(vals, native_pd.Series):
            values = pd.Series(vals)
        else:
            values = vals
        df.insert(1, column, values)
        return df
    with SqlCounter(query_count=query_count, join_count=1):
        eval_snowpark_pandas_result(
            snow_df, native_df, functools.partial(insert, "td", values)
        )

```

Reply:
I did the original way on purpose, it tries to also test the cases that original df contains a existing timedelta column or not and also test a chain of insert operations.